python/cudf/cudf/core/reshape.py

```diff
@@ -215,7 +230,30 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
     # when axis is 1 (column) we can concat with Series and Dataframes
     if axis == 1:
 
+        if join == "inner" and typ is cudf.Series:
+            new_df = cudf.concat(objs, axis=1)
+            indexes_list = []
+            for obj in objs:
+                indexes_list.append(cp.array(obj.index))
+            indexes = []
+            for i in range(len(new_df.index)):
+                for obj in indexes_list:
+                    if i not in obj:
+                        indexes.append(i)
+            df = new_df.drop(index=indexes)
```

Comment:
We're essentially recreating a join algorithm here but doing it via Python iteration which is going to be very slow.

Instead of doing this, we should be able to just use our existing `join` or `merge` implementations and do them pairwise. I.E. say I want to do `cudf.concat([a, b, c, d], axis=1, join="inner")` we could have an execution pattern that looks like (pseudocode):
```
result = a.join(b, on=index, how="inner")
result = result.join(c, on=index, how="inner")
result = result.join(d, on=index, how="inner")
return result
```

Reply:
Thank you, Keith! I'd thought about using join but didn't really know what to do if we had more than one series. I'll implement this :D

Reply:
I wonder if we could implement this: #6256 and use a single join for multiple columns?

Reply:
I'd say we implement here first and then move it underneath `join` in a follow up and use that within `concat`.